recipes/hexl/all/conanfile.py

```diff
@@ -0,0 +1,129 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+import os
+
+required_conan_version = ">=1.43.0"
+
+class HexlConan(ConanFile):
+    name = "hexl"
+    license = "Apache-2.0"
+    homepage = "https://github.com/intel/hexl"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Intel Homomorphic Encryption (HE) Acceleration Library"
+    topics = ("homomorphic", "encryption")
+    settings = "os", "compiler", "build_type", "arch"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "experimental": [True, False],
+        "fpga_compatibility_dyadic_multiply": [True, False],
+        "fpga_compatibility_keyswitch":  [True, False]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "experimental": False,
+        "fpga_compatibility_dyadic_multiply": False,
+        "fpga_compatibility_keyswitch": False
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("hexl-{0}".format(self.version), self._source_subfolder)
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.22.0")
+
+    def requirements(self):
+        self.requires("cpu_features/0.7.0")
+
+        if self.settings.build_type == "Debug":
+            self.requires("easyloggingpp/9.97.0")
+
+    def validate(self):
+        tools.check_min_cppstd(self, "17")
```

Comment:
```suggestion
    @property
    def _compilers_minimum_version(self):
        return {
            "gcc": "9",
            "Visual Studio": "15.7",
            "clang": "7",
            "apple-clang": "11",
        }

    def validate(self):
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, 17)
        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
        if minimum_version:
            if tools.Version(self.settings.compiler.version) < minimum_version:
                raise ConanInvalidConfiguration("{} requires C++17, which your compiler does not support.".format(self.name))
        else:
            self.output.warn("{} requires C++17. Your compiler is unknown. Assuming it supports C++17.".format(self.name))
```
Need to validate compiler version too.

Reply:
Are you sure for gcc minimum version ? Because mine is 7.5 and it compiles. Seems version 7 is enough : https://gcc.gnu.org/projects/cxx-status.html#cxx17

Reply:
Done, but gcc is configured as 7